pkg/controller/reconciler/deployment_reconciler.go

```diff
@@ -126,6 +126,7 @@ func (r *DeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	// If the Deployment is being deleted **or** if Radius is no longer enabled, then we should
 	// clean up any Radius state.
+	// FIXME: If the DeletionTimestamp is not nil, does it mean that Configuration and/or Status will also not be nil?
```

Comment:
This is possible in the following case:

- Deletion of the `Deployment` is in progress (`DeletionTimestamp != nil`)
- The `Deployment` is not using Radius `Configuration == nil`
- The `Deployment` was not previously using Radius `Status == nil`